gdal/ogr/ogrsf_frmts/cad/gdalcaddataset.cpp

```diff
@@ -0,0 +1,463 @@
+        {
+            char** papszPRJData = NULL;
+            papszPRJData = CSLAddString(papszPRJData, sESRISpatRef);
+            if( poSpatialRef->importFromESRI( papszPRJData ) != OGRERR_NONE )
+            {
+                CPLError( CE_Warning, CPLE_AppDefined,
+                        "Failed to parse PRJ section, ignoring." );
+                delete( poSpatialRef );
+                poSpatialRef = NULL;
+            }
+
+            CSLDestroy( papszPRJData );
+        }
+        else
+        {
+            const char * pszPRJFilename = GetPrjFilePath();
+            if(NULL != pszPRJFilename)
+            {
+            CPLPushErrorHandler( CPLQuietErrorHandler );
+            char **cabyPRJData = CSLLoad(pszPRJFilename);
+            CPLPopErrorHandler();
+
+            if( poSpatialRef->importFromESRI( cabyPRJData ) != OGRERR_NONE )
+            {
+                CPLError( CE_Warning, CPLE_AppDefined,
+                        "Failed to parse PRJ section, ignoring." );
+                delete( poSpatialRef );
+                poSpatialRef = NULL;
+            }
+
+            if(cabyPRJData)
+                CSLDestroy( cabyPRJData );
+        }
+    }
+    }
+
+    char *pszProjection = NULL;
+    poSpatialRef->exportToWkt( &pszProjection );
+    soWKT = pszProjection;
+    CPLFree( pszProjection );
+    return poSpatialRef;
+}
+
+const char* GDALCADDataset::GetPrjFilePath()
+{
+    const char * pszPRJFilename = CPLResetExtension(osCADFilename, "prj");
+    if ( CPLCheckForFile ((char*)pszPRJFilename, NULL) == TRUE)
+        return pszPRJFilename;
+    
+    pszPRJFilename = CPLResetExtension(osCADFilename, "PRJ");
+    if ( CPLCheckForFile ((char*)pszPRJFilename, NULL) == TRUE)
+        return pszPRJFilename;
+    
+    return NULL;
+}
+
+const char *GDALCADDataset::GetProjectionRef(void)
+{
+    return soWKT;
+}
+
+CPLErr GDALCADDataset::GetGeoTransform( double* padfGeoTransform )
+{
+    memcpy( padfGeoTransform, adfGeoTransform, sizeof(double) * 6 );
+    return CE_None;
+}
+                       
+int GDALCADDataset::GetGCPCount()
+{
+    if(NULL == poRasterDS)
+        return 0;
+    return poRasterDS->GetGCPCount();
+}
+
+const char *GDALCADDataset::GetGCPProjection()
+{
+    if(NULL == poRasterDS)
+        return NULL;
+    return poRasterDS->GetGCPProjection();    
+}
+
+const GDAL_GCP *GDALCADDataset::GetGCPs()
+{
+    if(NULL == poRasterDS)
+        return NULL;
+    return poRasterDS->GetGCPs();    
+} 
+
+char **GDALCADDataset::GetMetadataDomainList()
+{
+    if(NULL == poRasterDS)
+        return GDALDataset::GetMetadataDomainList();
+    return CSLMerge(GDALDataset::GetMetadataDomainList(), 
+                                    poRasterDS->GetMetadataDomainList());    
+}
+    
+char **GDALCADDataset::GetMetadata( const char * pszDomain  )
+{
+    if(NULL == poRasterDS)
+        return GDALDataset::GetMetadata(pszDomain);
+    return CSLMerge(GDALDataset::GetMetadata(pszDomain), 
```

Comment:
I think there is another problem here: each execution of GetMetadata the GDALDataset::pszDomain will increased. So I'll merge metadata of CAD file and proxy raster once in constructor. And remove  override GetMetadata, GetMetadataDomainList etc. as unnecessary.